02_bubble/sirigithub/main.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubbleSort(s []int) []int {
```

Comment:
Please make a copy of slice here.
In your case it's modifying the input slice via reference. Because of this 1st test is executed agains unsorted input, but 2nd, 3rd .. are executed against already sorted slice:
```
=== RUN   TestMainOutput
--- PASS: TestMainOutput (0.00s)
=== RUN   TestInsertionSort
Insertion input
{[] []}
Insertion input
{[1] [1]}
Insertion input
{[1 2 3 4 5] [1 2 3 4 5]}
Insertion input
{[10 9 8 7 6] [6 7 8 9 10]}
Insertion input
{[-10 9 8 7 6] [-10 6 7 8 9]}
Insertion input
{[3 2 1 5] [1 2 3 5]}
--- PASS: TestInsertionSort (0.00s)
=== RUN   TestBubbleSort
Bubble input
{[] []}
Bubble input
{[1] [1]}
Bubble input
{[1 2 3 4 5] [1 2 3 4 5]}
Bubble input
{[6 7 8 9 10] [6 7 8 9 10]}
Bubble input
{[-10 6 7 8 9] [-10 6 7 8 9]}
Bubble input
{[1 2 3 5] [1 2 3 5]}
--- PASS: TestBubbleSort (0.00s)
PASS
ok      _/tmp/ff        0.005s
```

Reply:
@n0npax good spot , totally overlooked that thank you

Reply:
@n0npax , fixed the code to work off a copy of the passed slice.